modules/ROOT/pages/_partials/sync/js-client-realtime-updates.adoc

```diff
@@ -1,3 +1,77 @@
 = Realtime Updates
 
-TODO
\ No newline at end of file
+Subscriptions are a core concept of GraphQL and one which allow us to enable real time updates of data
+from the server to the client.
+To do this GraphQL Subscriptions utilise websockets to enable clients to subscribe to published changes.
+
+The architecture of websockets is as follows:
+
+* Client connects to websocket server
+* Upon certain events, the server can publish the results of these events to the websocket
+* Any _currently connected_ client to that websocket will receive these results
+* The client can close the connection at any time and will no longer recieve updates
+
+Websockets are not however, a queueing system. For example if the client is offline when the server publish the changes
+the client will not receive them. To receive updates the client must be currently connected to the websocket server,
+updates made over this websocket while the client is offline will not be replayed.
+{sync-client} comes with subscription support out of the box including auto-reconnection upon device restart or network reconnect.
+To enable subscriptions on your client set the following
+paramater in the {sync-client} config object. A DataSyncConfig interface is also available from {sync-client} if you wish to use it.
+
+== Setting up a Client to use Subscriptions
+
+To set up a client to use subscriptions simply provide a wsUrl in your `mobile-services.json` file, {sync-client} will automatically
+apply the necessary logic to handle websocket requests. This can also be provided manually as seen below.
+
+```
+const config: DataSyncConfig {
+    ...
+    wsUrl: { WEB_SOCKET_URL_HERE }
+    ...
+}
+```
+Where wsUrl is the full URL of the websocket endpoint of your GraphQL server. Once this object is created, use it to initialise {sync-client} with:
+
+```
+import { createClient } from '@aerogear/voyager-client';
+
+const client = createClient(config)
+```
+
+== Using Subscriptions
+
+Once the client is configured we can then make use of subscriptions as follows.
+
+```
+
+getTasks() {
+  const tasks = client.watchQuery({
+    query: GET_TASKS
+  });
+
+  tasks.subscribeToMore({
+    document: TASK_ADDED_SUBSCRIPTION,
+    updateQuery: (prev, { subscriptionData }) => {
+      if(subscribtionData.data){
+        const newTask = subscribtionData.data.taskAdded;
+        return allTasks: [
+            ...prev,
+            newItem
+          ];
+        } else {
+          return prev
+        }
+    }
+  });
+  return tasks;
```

Comment:
do we need to add some comments in the code to explain what the code does?

Reply:
We need to say why we use subscribeToMore over regular subscription (cache updates).
And entire pattern when using subscriptions:

1) Make network query to get fresh data
2) Watch cache changes
3) Subscribe
4) Unsubscribe when leaving view (important as can lead to memory leaks).
We do not do that because we had single view, now we have entire app so we may do that as well to not subscribe many times.